install/v2plugin/startcontiv.sh

```diff
@@ -4,125 +4,111 @@
+             --log-file=/var/log/contiv/ovs-db.log -vsyslog:info -vfile:info \
+             --pidfile --detach /etc/openvswitch/conf.db
+
+echo "INFO: Starting ovs-vswitchd"
+ovs-vswitchd -v --pidfile --detach --log-file=/var/log/contiv/ovs-vswitchd.log \
+    -vconsole:err -vsyslog:info -vfile:info &
+
+retry=0
+while [[ $(ovsdb-client list-dbs | grep -c Open_vSwitch) -eq 0 ]] ; do
+    if [[ ${retry} -eq 5 ]]; then
+        echo "CRITICAL: Failed to start ovsdb in 5 seconds."
+        exit 1
+    else
+        echo "INFO: Waiting for ovsdb to start..."
+        sleep 1
+        ((retry+=1))
+    fi
+done
 
-echo "  Starting OVSBD server " >> $BOOTUP_LOGFILE
-ovsdb-server --remote=punix:/var/run/openvswitch/db.sock --remote=db:Open_vSwitch,Open_vSwitch,manager_options --private-key=db:Open_vSwitch,SSL,private_key --certificate=db:Open_vSwitch,SSL,certificate --bootstrap-ca-cert=db:Open_vSwitch,SSL,ca_cert --log-file=$log_dir/ovs-db.log -vsyslog:dbg -vfile:dbg --pidfile --detach /etc/openvswitch/conf.db >> $BOOTUP_LOGFILE
-echo "  Starting ovs-vswitchd " >> $BOOTUP_LOGFILE
-ovs-vswitchd -v --pidfile --detach --log-file=$log_dir/ovs-vswitchd.log -vconsole:err -vsyslog:info -vfile:info &
+echo "INFO: Setting OVS manager (tcp)..."
 ovs-vsctl set-manager tcp:127.0.0.1:6640
-ovs-vsctl set-manager ptcp:6640
 
-echo "Started OVS, logs in $log_dir" >> $BOOTUP_LOGFILE
+echo "INFO: Setting OVS manager (ptcp)..."
+ovs-vsctl set-manager ptcp:6640
 
+# starting services
 set +e
-
-if [ $plugin_role == "master" ]; then
-    if [ -z "$fwd_mode" ]; then
-        echo "fwd_mode is not set, plugin cannot be enabled"
-        exit 1
-    fi
-    echo "Starting Netmaster " >> $BOOTUP_LOGFILE
+if [ "$CONTIV_ROLE" = "netmaster" ]; then
     while  true ; do
-        echo "/netmaster $dbg_flag -plugin-name=$plugin_name -cluster-mode=$plugin_mode -cluster-store=$cluster_store $listen_url_cfg $control_url_cfg" >> $BOOTUP_LOGFILE
-        /netmaster $dbg_flag -plugin-name=$plugin_name -cluster-mode=$plugin_mode -cluster-store=$cluster_store $listen_url_cfg $control_url_cfg &> $log_dir/netmaster.log
-        echo "CRITICAL : Net Master has exited, Respawn in 5s" >> $BOOTUP_LOGFILE
-	mv $log_dir/netmaster.log $log_dir/netmaster.log.lastrun
+        echo "INFO: Starting contiv netmaster"
+        set -x
+        /contiv/bin/netmaster "$@" &> "$CONTIV_LOG_DIR/netmaster.log"
+        set +x
+        echo "ERROR: Contiv netmaster has exited, restarting in 5s"
         sleep 5
-        echo "Restarting Netmaster " >> $BOOTUP_LOGFILE
     done &
-
-    set -e
-    echo "Waiting for netmaster to be ready for connections"
-    # wait till netmaster starts to listen
-    for i in $(seq 1 10); do
-        [ "$(curl -s -o /dev/null -w '%{http_code}' $control_url)" != "000" ] \
-           && break
-        sleep 1
-    done
-    if [ "$i" -ge "10" ]; then
-        echo "netmaster port not open (needed to set forwarding mode), plugin failed"
-        exit 1
-    fi
```

Comment:
why are we getting rid of the wait for the listen port?

Reply:
b/c the fwd config are in passed CLI, don't have to wait for it's ready

Reply:
k